src/lib/chips/chip-list.ts

```diff
@@ -3,17 +3,31 @@ import {
   ChangeDetectionStrategy,
   Component,
   ContentChildren,
+  Directive,
   Input,
   QueryList,
   ViewEncapsulation,
   OnDestroy,
+  ElementRef,
+  Renderer2,
 } from '@angular/core';
 
 import {MdChip} from './chip';
 import {FocusKeyManager} from '../core/a11y/focus-key-manager';
 import {coerceBooleanProperty} from '../core/coercion/boolean-property';
-import {SPACE, LEFT_ARROW, RIGHT_ARROW, TAB} from '../core/keyboard/keycodes';
 import {Subscription} from 'rxjs/Subscription';
+import {
+  LEFT_ARROW, RIGHT_ARROW, SPACE, BACKSPACE, DELETE, UP_ARROW, DOWN_ARROW
+} from '../core/keyboard/keycodes';
+import {Dir} from '../core/rtl/dir';
+
+@Directive({
+  selector: '[mdChipListContainer], [matChipListContainer]',
```

Comment:
One way to avoid this directive is 
```
<md-input-container>
  <md-chip-list>
    <input mdInput mdChipList />
  </md-chip-list>
</md-input-container>
```